File: Backend/Backend/Model/VehicleServiceLog.cs
using System;
using System.Collections.Generic;

namespace Backend.Model;

public partial class VehicleServiceLog
{
    public int ServiceId { get; set; }

    public int VehicleId { get; set; }

    public string? ServiceDetails { get; set; }

    public DateTime? LastDate { get; set; }

    public DateTime? DueDate { get; set; }
}

File: Frontend/src/app/model/vehicleServiceLog.ts
export class VehicleServiceLog{
    public serviceId: number;
    public vehicleId: number;
    public serviceDetails: string;
    public LastDate: Date;
    public DueDate: Date;

    constructor(serviceId: number, vehicleId: number, serviceDetails: string, LastDate: Date, DueDate:Date){
        this.serviceId = serviceId;
        this.vehicleId = vehicleId;
        this.serviceDetails = serviceDetails;
        this.LastDate = LastDate;
        this.DueDate = DueDate;
    }
}

export interface IVehicleServiceLog{
    serviceId: number;
    vehicleId: number;
    serviceDetails: string;
    LastDate: Date;
    DueDate: Date;
}